components/entities/api/src/main/java/org/phenotips/entities/PrimaryEntity.java

```diff
@@ -53,7 +55,14 @@
      *
      * @return a valid document reference
      */
-    DocumentReference getDocument();
+    DocumentReference getDocumentReference();
+
+    /**
+     * Returns the document where the entity is stored.
+     *
+     * @return a valid document reference
+     */
+    XWikiDocument getDocument();
```

Comment:
-1 for changing the return type, only because this method is also exposed in the old `Patient` interface, which is usable not just from Java code, but also from scripts. While we can detect broken Java code at compile time, there's no check for Velocity.

Another thing to keep in mind: `Patient` is returned by `$services.patients`, but the internal `com.xpn.xwiki.document.XWikiDocument` should __never__ be exposed in scripts, since this one bypasses all permission checks. So `SecurePatientRepository` should return a new `Patient` implementation which returns `null` whenever `getXWikiDocument` is invoked.